### DISCUSSION TITLE
Jax  jacfwd() vs numeric graident

### DISCUSSION THREAD

**Discussion by RaviPandey33** [jax-ml/jax#22057] [Q&A]
I am trying to find gradient for this 
<img width="561" alt="image" src="https://github.com/google/jax/assets/10897765/03869b83-0a2f-4480-83c6-d5fd61027292">

Both the methods , jacfwd() and numeric gradient gives me different answers. Could anyone tell me the reason ? 
Is it because of the use of jnp.abs() value ? 

which one should i consider correct ?

**Comment by jakevdp** [comment]
It's hard to say for sure without knowing the inputs you're using, or how you're computing the numerical gradient. But if I had to guess, I suspect some of your input values are near the discontinuity at zero, and your numerical integration offsets are straddling that discontinuity. If that's the case, then the analytic gradient is certainly more accurate.
